misc/pragmas.md

```diff
@@ -0,0 +1,574 @@
+helpful to render a grammatical attribute as an XML attribute.)
+
+*Are there other use cases that need to be mentioned here?*
+
+Note that some of these use cases may in practice be handled by
+changes to the syntax of ixml. We include them in the list of use
+cases for pragmas not because we think they are best handled by
+pragmas (although for some of these cases, that may be true) but
+because they are (a) plausible ideas for things one might want to do
+which are (b) not supported by ixml in its current form, and thus (c)
+natural examples of the kinds of things an extension mechanism oughtly
+ideally to support.
+
+Some of these use cases seem most naturally handled by annotations
+which apply to a grammar as a whole, some by annotations which apply
+to individual rules, and some by annotations which apply to individual
+symbols in the grammar.
+
+We do not see a strong use case for annotations which apply to
+arbitrary expressions in a grammar.
+
+## Requirements and desiderata
+
+Our tentative list of requirements and desiderata is as follows.
+
+By *requirement* we mean a property or functionality which must be
+achieved for a pragmas proposal to be worth adopting.  
+By *desideratum* we mean a property or functionality that should be
+included if possible, but which need not lead to the rejection of the
+proposal if it proves impossible to achieve.
+
+Requirements:
+
+* It must be straightforward for processors to ignore pragmas they do 
+not understand, and to determine whether they `understand' a given 
+pragma or not. 
+
+* It must be clear to human readers and software which expressions in
+standard ixml notation are and are not affected or overridden by a
+given pragma.
+
+* Any pragma must thus specify (explicitly or implicitly) both what
+should be done by a processor that understands and processes the
+pragma and what should be done by a processor that does not understand
+and process the pragma. We refer to the latter as the *fallback
+expression*.
+
+
+Desiderata:
+
+* Ideally, the result of evaluating the fallback expression should be
+a useful and meaningful result, but this is more a matter for the
+individual writing a grammar than for this proposal.
+
+* It should ideally be possible to specify pragmas as annotations
+applying to a symbol, a rule, or a grammar as a whole, and it should
+be possible to know which is which. It is not required that the
+distinction be a syntactic one, however, since it can also be
+expressed by the semantics of the particular pragma.
+
+## Design questions
+
+Several design questions can be distinguished, not necessarily
+completely orthogonal.
+
+* What syntax should pragmas have in invisible XML? 
+
+* What representation should pragmas have in the XML serialization of
+an ixml grammar?
+
+* Where can pragmas appear?
+
+
+## Proposal(s)
+
+Two concrete proposals have been made; they have been given arbitrary
+names for discussion.
+
```

Comment:
As the proposer of the hash quotes proposal, I would like to drop it in favour of the hash-QName proposal.  Once we identify the need for a number of parameters, and thus the square brackets, I think that it doesn't make sense to have another, less powerful way of expressing pragma.

I think there's also an issue/scope for confusion where we have a pragma in front of a string!